core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -529,6 +555,24 @@ public TableMetadata replaceProperties(Map<String, String> rawProperties) {
         .build();
   }
 
+  public TableMetadata replaceRefs(Map<String, SnapshotRef> newRefs) {
```

Comment:
I don't think that replacing all refs is a good idea. In fact, I would probably avoid adding new mutation methods to `TableMetadata` in general and have you go directly to the `TableMetadata.Builder` API.